avm/res/key-vault/vault/access-policy/main.bicep

```diff
@@ -6,13 +6,13 @@ metadata owner = 'Azure/module-maintainers'
 param keyVaultName string
 
 @description('Optional. An array of 0 to 16 identities that have access to the key vault. All identities in the array must use the same tenant ID as the key vault\'s tenant ID.')
-param accessPolicies array?
+param accessPolicies accessPoliciesType
 
 var formattedAccessPolicies = [for accessPolicy in (accessPolicies ?? []): {
-  applicationId: contains(accessPolicy, 'applicationId') ? accessPolicy.applicationId : ''
-  objectId: contains(accessPolicy, 'objectId') ? accessPolicy.objectId : ''
+  applicationId: accessPolicy.?applicationId ?? ''
+  objectId: accessPolicy.?objectId ?? ''
```

Comment:
This is an interesting one @johnlokerse - in the loop you re-use the original logic that treats the `objectId` as optional.
Yet in the UDT you defined it as required. Which is it? :D 

As per the [API Specs](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2023-07-01/keyvault.json#L1063-L1091):
```JSON
"AccessPolicyEntry": {
  "properties": {
    "tenantId": {
      "type": "string",
      "format": "uuid",
      "description": "The Azure Active Directory tenant ID that should be used for authenticating requests to the key vault."
    },
    "objectId": {
      "type": "string",
      "description": "The object ID of a user, service principal or security group in the Azure Active Directory tenant for the vault. The object ID must be unique for the list of access policies."
    },
    "applicationId": {
      "type": "string",
      "format": "uuid",
      "description": " Application ID of the client making request on behalf of a principal"
    },
    "permissions": {
      "$ref": "#/definitions/Permissions",
      "description": "Permissions the identity has for keys, secrets and certificates."
    }
  },
  "description": "An identity that have access to the key vault. All identities in the array must use the same tenant ID as the key vault's tenant ID.",
  "required": [
    "tenantId",
    "objectId",
    "permissions"
  ],
  "type": "object"
}
```
I'd argue the loop must be updated to
```suggestion
  objectId: accessPolicy.objectId
```

Reply:
Note: I advice to re-run the `Set-AVMModule` utility as this will require an update of the `main.json` files

Reply:
Yes, correct 😄... Fixed and ran Set-AVMModule again.